ProcessLib/ThermoMechanics/ThermoMechanicsFEM.h

```diff
@@ -246,11 +253,12 @@ class ThermoMechanicsLocalAssembler
                     DisplacementDim>::value>;
 
             // assume isotropic thermal expansion
+            const double T_ip = N.dot(T);  // T at integration point
             eps_m.noalias() =
                 eps - linear_thermal_strain * Invariants::identity2;
             auto&& solution = _ip_data[ip].solid_material.integrateStress(
-                t, x_position, dt, eps_m_prev, eps_m, sigma_prev, *state,
-                delta_T + _process_data.reference_temperature);
+                t, x_position, dt, eps_m_prev, eps_m, sigma_prev, *state, T_ip);
+            eps_m.noalias() = eps;
```

Comment:
This assignment seems to overwrite the assignment in the line above. Is that intended?

Reply:
Yes, it is intended.  Inside integrateStress, eps_m and  eps_m_prev is used to calculate d eps^e such that
``d eps^e =  eps_m - eps_m_prev``. In the previous implementation,   ``eps_m = B u^{n+1} - \alpha_T (T^{n+1}-T_{ref}) I``and  ``eps_m_prev = B u^{n} - \alpha_T (T^{n}-T_{ref}) I`` (by pushBackState()). With the current change,    ``eps_m = B u^{n+1} - \alpha_T dT I``, and ``eps_m_prev = B u^{n}``. In order to update ``eps_m_prev = B u^{n}`` by pushBackState(), we have to have the assignment.

This is also the reason that I mentioned in the introduction that eps_m and eps_m_prev can removed from _ip_data even under large deformation. We only need the strain increment in the stress integration. Since we already have access to displacement and the increment of displacement,  the strain increment can be computed directly without storing two step strains.

Reply:
Thanks for the explanation. For me (as a non-expert), the explanation would also be very helpful directly in the code. But maybe people who know the topic don't need it, and maybe it's already said in some PDF explaining the governing equations.

Basically, the cause for my confusion was that I read the lines 257-260 wrongly. I thought, they were one statement, but actually they are two. Thanks anyway.

> This is also the reason that I mentioned in the introduction that eps_m and eps_m_prev can removed from _ip_data even under large deformation. We only need the strain increment in the stress integration. Since we already have access to displacement and the increment of displacement, the strain increment can be computed directly without storing two step strains.

I thought, the design choice was plasticity-related. But maybe I'm wrong.

Reply:
For plastic model, we only need the total strain increment to be passed into integrateStress. Of course, I would not change the code for this before a discussion.